### PR TITLE
[BIT] Warp old torch convert rule with Code() in rules.py

### PR DESCRIPTION
**Pull request by Cutelemon6**
更新以下 Rule 的旧版转换规则，用 Code() 封装 impl

```
__rshift__Rule
__Pow__Rule
ScatterndaddRule
ScatterndRule
ScatterRule
NumelRule
LcmRule
ExpandasRule
ExpandRule
BroadcastShapeRule
```

在 accuracy 模式复测了使用以上 rule 的位于 5_accuracy 目录下的 api 配置，均通过

```
paddle.Tensor.__pow__
paddle.Tensor.__rshift__
paddle.bitwise_right_shift
paddle.scatter
paddle.numel
paddle.lcm
paddle.expand_as
paddle.Tensor.expand
paddle.expand
paddle.nn.functional.dropout
paddle.broadcast_shape
```

### CODE REVIEW

File: tester/paddle_to_torch/rules.py [hunk 1]
```diff
@@ -608,12 +608,14 @@ def apply(self, paddle_api: str) -> ConvertResult:
 
 class BroadcastShapeRule(BaseRule):
     def apply(self, paddle_api: str) -> ConvertResult:
-        impl = """
+        pre = """
 x_shape = locals().get('x_shape')
 y_shape = locals().get('y_shape')
+"""
+        core = """
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:615]: core 可以改成：
```
core = "result = torch.broadcast_shapes(x_shape, y_shape)"
```

File: tester/paddle_to_torch/rules.py [hunk 2]
```diff
@@ -1501,45 +1503,55 @@ def axis_dropout(x, p, axis, training=True, mode='upscale_in_train'):
 axis = locals().get('axis')
 training = locals().get('training')
 mode = locals().get('mode')
+"""
+        core = """
 result = axis_dropout(x, p, axis, training, mode) if axis is not None else torch.nn.functional.dropout(input=x, p=float(p), training=training)
 """
-        code = impl.splitlines()
-        return ConvertResult.success(paddle_api, code, "result")
+        code = Code(preprocess=pre.splitlines(), core=[core])
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:1510]: 多行 core 使用 .splitlines()

File: tester/paddle_to_torch/rules.py [hunk 3]
```diff
@@ -1501,45 +1503,55 @@ def axis_dropout(x, p, axis, training=True, mode='upscale_in_train'):
...
 p = locals().get('p')
 training = locals().get('training')
 data_format = locals().get('data_format')
 
 if data_format == "NDHWC":
     x = x.permute(0, 4, 1, 2, 3)
+"""
+        core = """
 result = torch.nn.functional.dropout3d(input=x, p=float(p), training=training)
+"""
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:1549]: +1

File: tester/paddle_to_torch/rules.py [hunk 4]
```diff
@@ -6315,18 +6327,20 @@ def apply(self, paddle_api: str) -> ConvertResult:
 # __
 class __Pow__Rule(BaseRule):
     def apply(self, paddle_api: str) -> ConvertResult:
-        impl = """
+        pre = """
 tensor = locals().get('self')
 other = locals().get('y')
+"""
+        core = """
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:6334]: +1

File: tester/paddle_to_torch/rules.py [hunk 5]
```diff
@@ -6340,15 +6354,16 @@ def logical_right_shift(x: torch.Tensor, y: torch.Tensor):
...
 if is_arithmetic:
     result = tensor.__rshift__(other)
 else:
     # logical right shift 
     result = logical_right_shift(tensor, other)
 """
-        code = impl.splitlines()
-        return ConvertResult.success(paddle_api, code)
+        code = Code(preprocess=pre.splitlines(), core=[core])
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:6365]: +1

**Review by cangtianhuang — CHANGES_REQUESTED** [review]
